src/ripple/app/consensus/RCLValidations.cpp

```diff
@@ -240,24 +240,45 @@ RCLValidations::onStale(RCLValidation&& v)
         return;
```

Comment:
The intent is that `onStale` is only accessible and callable from the generic `Validations` code and any calls into the `Validations` code from `RCLConsensus` require holding the lock for the duration of the call to the base class.  The fact that this is opaque is a shortcoming of the CRTP design.  I'll try to think of a way to make it clearer outside of just commenting.  I'm open to suggestions too =).